packages/typespec-ts/test/integration/typespec/union-body/main.tsp

```diff
@@ -28,9 +28,9 @@ model ReturnResponse {
 namespace NicepayPlatform {
   @post
   @route("/request-union-body")
-  op requestUnionBody(@body request: RequestRegisterCC | RequestRegisterVA): {};
+  op requestUnionBody(@body request: RequestRegisterCC | RequestRegisterVA): {@bodyRoot _ : {};};
```

Comment:
what does the original {} gives us? void?

Reply:
void.

Reply:
shoudl be `@body` here if you want actually `{}`, that is actually I think a bug, `@bodyRoot` should just move the body resolution down to that property instead of the root of the response type and not change the meaning

Reply:
filed https://github.com/microsoft/typespec/issues/3372

Reply:
can we use `{@bodyRoot _: void }` to express void body? I feel like it's strange if both `{@bodyRoot _: void }` and `{@bodyRoot _: {} }` means response body void? 
Also, I am curious if we can use bodyRoot to express non-object and non-void type body, like string, boolean etc with {}

Reply:
The logic is as follow
1. `@bodyRoot` should just move the implicit body resolution from the root of the request/response to the property you have it. So it should be exactly the same handling to have `op foo(): X` vs `op foo(): {@bodyRoot _: X}` as the over the write meaning.
2. Now the reason why `{}` means no body is because `{@header foo: string}` means no body but `{}` didn't use to. If a property was removed with visibility it wasn't meaning empty body either unless it was marked with `@header` as well. So it was very inconsitent. So to simplify the logic is: after removing the non applicable properties if there is none left it means we have no body. In the rare case where you actually want to send `{}` you can use `@body` which will expect it to be the exact body.

Reply:
I do understand the intention for bodyRoot as I read about this before when I was working on visibility related design spec, I was just feel like since void can't be used in property type, maybe we should not allow `{@bodyRoot _: void }` too?